adapters/repos/db/vector/flat/index.go

```diff
@@ -428,7 +434,12 @@ func (index *flat) createDistanceCalc(vector []float32) distanceCalc {
 		defer index.pool.float32SlicePool.Put(vecSlice)
 
 		candidate := float32SliceFromByteSlice(vecAsBytes, vecSlice.slice)
-		return index.distancerProvider.SingleDist(vector, candidate)
+		dist, err := index.distancerProvider.SingleDist(vector, candidate)
+		if err != nil {
+			return dist, err
+		}
+
+		return dist, nil
```

Comment:
```suggestion
		return index.DistanceBetweenVectors(vector, candidate)
```
Should this re-use [DistanceBetweenVectors](https://github.com/weaviate/weaviate/pull/7875/files#diff-f4246823075bd093c22660de1de0ad72c74406f5416b82c7054994e7bd74a76dR897)?